dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java

```diff
@@ -267,12 +300,44 @@ public DependResult getModelDependResult(Date currentTime, int testFlag) {
             DependResult dependResult = getDependResultForItem(dependentItem, currentTime, testFlag);
             if (dependResult != DependResult.WAITING && dependResult != DependResult.FAILED) {
                 dependResultMap.put(dependentItem.getKey(), dependResult);
+                if (dependentItem.getParameterPassing() && !dependItemVarPoolPropertyMap.isEmpty()) {
+                    addTaskVarPool(dependItemVarPoolPropertyMap, dependItemVarPoolEndTimeMap,
+                            dependTaskVarPoolPropertyMap, dependTaskVarPoolEndTimeMap);
+                }
             }
+            dependItemVarPoolPropertyMap.clear();
+            dependItemVarPoolEndTimeMap.clear();
```

Comment:
Do we really have to clear these two maps?

Reply:
Oh, I'm not familiar with dependent task logic, but if you make sure there's no concurrent issue, I'm ok with that.

Reply:
In the `DependentExecute` class, I think the function execution should be parallel